test/e2e/list_diff_on_tags.rb

```diff
@@ -0,0 +1,80 @@
+# rubocop:disable all
+
+require_relative "../e2e"
+require 'yaml'
+
+pipeline = %{
+version: v1.0
+name: Test
+agent:
+  machine:
+    type: e1-standard-2
+
+blocks:
+  - name: Test
+    run:
+      when: "branch = 'master' and change_in(['/app', 'log.txt'], {on_tags: true})"
+    task:
+      jobs:
+        - name: Hello
+          commands:
+            - echo "Hello World"
+
+  - name: Test2
+    run:
+      when: "branch = 'master' and change_in('/app', {on_tags: false})"
+    task:
+      jobs:
+        - name: Hello
+          commands:
+            - echo "Hello World"
+
+  - name: Test3
+    run:
+      when: "branch = 'master' and change_in('/app')"
+    task:
+      jobs:
+        - name: Hello
+          commands:
+            - echo "Hello World"
+}
+
+origin = TestRepoForChangeIn.setup()
+
+origin.add_file('.semaphore/semaphore.yml', pipeline)
+origin.commit!("Bootstrap")
+
+origin.create_branch("dev")
+origin.add_file("lib/B.txt", "hello")
+origin.commit!("Changes in dev")
+
+repo = origin.clone_local_copy(branch: "dev")
+
+repo.run(%{
```

Comment:
Modified tags use-case to behave like that + removed redundant `on-tags` argument from `list-diff` (not used, defaulted it to `true`).
Let me know if that'll work.